packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -181,66 +184,71 @@ export class SummaryManager extends TypedEventEmitter<ISummaryManagerEvents> imp
     };
 
     private startSummarization() {
+        assert(this.state === SummaryManagerState.Off, "expected: off");
         this.state = SummaryManagerState.Starting;
 
-        // throttle creation of new summarizer containers to prevent spamming the server with websocket connections
-        const delayMs = this.startThrottler.getDelay();
-        if (delayMs > 0 && delayMs >= this.startThrottler.maxDelayMs) {
-            // we can't create a summarizer for some reason; raise error on container
-            this.emit(
-                "summarizerWarning",
-                createSummarizingWarning("SummaryManager: CreateSummarizer Max Throttle Delay", false),
-            );
-        }
+        assert(this.runningSummarizer === undefined, "Old summarizer is still working!");
+
+        this.delayBeforeCreatingSummarizer().then(async (isDelayed: boolean) => {
+            // Re-validate that it need to be running. Due to asynchrony, it may be not the case anymore
+            // but only if creation was delayed. If it was not, then we want to ensure we always create
+            // a summarizer to kick off lastSummary. Without that, we would not be able to summarize and get
+            // document out of broken state if it has too many ops and ordering service keeps nacking main
+            // container (and thus it goes into cycle of reconnects)
+            if (isDelayed && this.getShouldSummarizeState().shouldSummarize === false) {
```

Comment:
`isDelay` doesn't truly capture what the intent is since whether this is true or false, it may have been delayed. Maybe `shouldDelayCreation` or `shouldStartImmediately` would capture the intent more clearly.